exercises/palindrome-products/test/test_palindrome_products.c

```diff
@@ -9,24 +12,45 @@ void tearDown(void)
 {
 }
 
-void check_factors(factor_t * actual, int depth, factor_t expected[])
+bool factor_t_are_equal(const factor_t * const f1, const factor_t * const f2)
+{
+   return ((f1->factor_a == f2->factor_a) && (f1->factor_b == f2->factor_b)) ||
```

Comment:
You can decide here, but it seems a lot clearer to me to use min/max to do this, but that's just me.

```
return (min(f1->factor_a, f1->factor_b) == min(f2->factor_a, f2->factor_b)) && 
       (max(f1->factor_a, f1->factor_b) == max(f2->factor_a, f2->factor_b));
```

Reply:
Ugh, nevermind. I didn't realize min/max aren't in the C stdlib, so you'd have to define them too.